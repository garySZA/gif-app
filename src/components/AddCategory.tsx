import React, { useState } from "react";
import { AddCategoryProps } from '../types/componentTypes';

export const AddCategory:React.FC<AddCategoryProps> = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState<string>('');
    
    const onInputChange = (event:any):void => {
        
        setInputValue(event.target.value);
    }

    const onSubmit = (event:any):void => {
        event.preventDefault();

        const value:string = inputValue.trim();

        if( value.length <= 1 ) return;
        onNewCategory( value );
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input
                type="text"
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
        )

}
