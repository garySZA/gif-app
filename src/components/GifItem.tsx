import React from "react"
import { GifItemProps } from "../types/componentTypes"

export const GifItem: React.FC<GifItemProps> = ({ image: { title, url } }) => {
    
    return (
        <div className="card">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}
