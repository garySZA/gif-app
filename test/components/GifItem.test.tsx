import React from "react";
import { getByRole, render, screen } from "@testing-library/react"
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en el componente GifItem', () => { 
    const image = {
        id: '123123',
        title: 'este es un titulo',
        url: 'https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=ec5732f0b50oz49jkhdlxakc2ju59epei8s23pfy343qm5wj&ep=v1_gifs_search&rid=giphy.gif&ct=g'
    };

    test('Debe de hacer match con el snapshot', () => {
        
        const { container } = render( <GifItem image={ image }/> )

        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el URL indicado', () => {
        render( <GifItem image={ image }/> )
        // screen.debug();

        const img = screen.getByRole('img') as HTMLElement;

        expect( img.getAttribute('src') ).toBe( image.url );

    });

    test('Debe de mostrar la imagen con el ALT indicado', () => {
        render( <GifItem image={ image }/> )
        // screen.debug();

        const img = screen.getByRole('img') as HTMLElement;

        expect( img.getAttribute('alt') ).toBe( image.title );

    });

    test('Debe de mostrar el titulo en el componente', () => { 
        render(<GifItem image={image} />);

        expect( screen.getByText( image.title ) ).toBeTruthy();
    });
});