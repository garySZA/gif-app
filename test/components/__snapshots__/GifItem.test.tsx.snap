// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="este es un titulo"
      src="https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=ec5732f0b50oz49jkhdlxakc2ju59epei8s23pfy343qm5wj&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      este es un titulo
    </p>
  </div>
</div>
`;
